src/features/members/membersSlice.js

```diff
@@ -41,6 +50,25 @@ export const addMember = createAsyncThunk(
   },
 );
 
+export const importMembers = createAsyncThunk(
+  'members/importMembers',
+  async (args) => {
+    const { cohort, emailList } = args;
+    const client = getAuthenticatedHttpClient();
+    const baseUrl = getConfig().LMS_BASE_URL;
+    const { data } = await client.post(
+      `${baseUrl}/api/partnerships/v0/memberships/${cohort}/`,
+      snakeCaseObject(emailList.map(email => ({ email }))),
+    );
+    const newMembers = data.map(member => {
```

Comment:
I totally missed that the backend wasn't returning IDs. Is it possible to get real IDs from the API instead of making temporary ones here?

Reply:
The serializer is returning null IDs for new members, which I had thought was intended since these memberships are not yet associated with users, but I'll double check that we aren't erroneously creating entries without IDs.

Reply:
Ah, I think it's due to this:
> If the model’s primary key is an [AutoField](https://docs.djangoproject.com/en/5.0/ref/models/fields/#django.db.models.AutoField) and ignore_conflicts is False, the primary key attribute can only be retrieved on certain databases (currently PostgreSQL, MariaDB 10.5+, and SQLite 3.35+). On other databases, it will not be set.

I wonder if using `bulk_create` is necessary yet, I suspect we'd really only need that for really large lists.

Reply:
Yeah, I was thinking the same, I think that should be fine. Might want an index on `mogc_partnerships_cohortmembership.email` eventually.